apps/system/lockscreen/js/lockscreen.js

```diff
@@ -851,15 +852,15 @@
    */
   LockScreen.prototype.checkPassCode =
   function lockscreen_checkPassCode(passcode) {
-    var request = {
-      passcode: passcode,
-      onsuccess: this.onPasscodeValidationSuccess.bind(this),
-      onerror: this.onPasscodeValidationFailed.bind(this),
-    };
-    window.dispatchEvent(new CustomEvent(
-      'lockscreen-request-passcode-validate',
-      { detail: request }
-    ));
+    PasscodeHelper.check(passcode).then((result) => {
```

Comment:
OK. I think it's good. Let's do this according to the design. And maybe it should be "Device passcode", not "ScreenLock passcode" anymore. Anyway, let's land the change first.